### PR TITLE
feat: partial fit

### PR DESCRIPTION
**Pull request by davidbp**

### CODE REVIEW

File: pqlite/index.py [hunk 1]
```diff
@@ -74,6 +76,7 @@ def __init__(
         self._use_smart_probing = True
 
         self.read_only = read_only
+        self.logger = logger_flag
```
Comment (numb3r3) [inline pqlite/index.py:79]: I would reject this change about logging. For the logging, we should use logging API `logger.setLevel()` to turn off debug/info message.

File: pqlite/index.py [hunk 2]
```diff
@@ -82,22 +85,27 @@ def __init__(
...
         self.vq_codec = None
         if self._vq_codec_path.exists() and n_cells > 1:
-            logger.info(
-                f'Load trained VQ codec (K={self.n_cells}) from {self.model_path}'
-            )
+
+            if self.logger:
+                logger.info(
+                    f'Load trained VQ codec (K={self.n_cells}) from {self.model_path}'
+                )
```
Comment (numb3r3) [inline pqlite/index.py:92]: If you don't want report this message. we can change the message level via `logger.debug()` instead.

File: pqlite/index.py [hunk 3]
```diff
@@ -82,22 +85,27 @@ def __init__(
...
             self.vq_codec = VQCodec.load(self._vq_codec_path)
         elif n_cells > 1:
-            logger.info(f'Initialize VQ codec (K={self.n_cells})')
+            if self.logger:
+                logger.info(f'Initialize VQ codec (K={self.n_cells})')
```
Comment (numb3r3) [inline pqlite/index.py:96]: same as above

File: pqlite/index.py (hunk 1 above)
Comment (numb3r3) [reply pqlite/index.py:79]: Now, the default level is `DEBUG`, we can change default as `INFO`

File: pqlite/index.py [hunk 4]
```diff
@@ -134,36 +142,75 @@ def train(self, x: np.ndarray, auto_save: bool = True, force_retrain: bool = Fal
         n_data, _ = self._sanity_check(x)
 
         if self.is_trained and not force_retrain:
-            logger.warning(
-                'The pqlite has been trained or is not trainable. Please use ``force_retrain=True`` to retrain.'
-            )
+            if self.logger:
+                logger.warning(
+                    'The pqlite has been trained or is not trainable. Please use ``force_retrain=True`` to retrain.'
+                )
```
Comment (numb3r3) [inline pqlite/index.py:148]: same

File: pqlite/index.py [hunk 5]
```diff
@@ -134,36 +142,75 @@ def train(self, x: np.ndarray, auto_save: bool = True, force_retrain: bool = Fal
...
             return
 
         if self.vq_codec:
-            logger.info(
-                f'Start training VQ codec (K={self.n_cells}) with {n_data} data...'
-            )
+            if self.logger:
+                logger.info(
+                    f'Start training VQ codec (K={self.n_cells}) with {n_data} data...'
+                )
```
Comment (numb3r3) [inline pqlite/index.py:155]: same

**Review by numb3r3 — CHANGES_REQUESTED** [review]

File: pqlite/index.py (hunk 1 above)
Comment (davidbp) [reply pqlite/index.py]: Removed the logger flag

**Pull request merged**